super-csv/src/main/java/org/supercsv/util/ReflectionUtils.java

```diff
@@ -208,23 +209,24 @@ public static Method findSetter(final Object object, final String fieldName, fin
 	private static Method findSetterWithCompatibleParamType(final Class<?> clazz, final String setterName,
 		final Class<?> argumentType) {
 		
-		Method compatibleSetter = null;
-		for( final Method method : clazz.getMethods() ) {
-			
-			if( !setterName.equalsIgnoreCase(method.getName()) || method.getParameterTypes().length != 1 ) {
-				continue; // setter must have correct name and only 1 parameter
-			}
-			
-			final Class<?> parameterType = method.getParameterTypes()[0];
-			if( parameterType.equals(argumentType) ) {
-				compatibleSetter = method;
-				break; // exact match
-				
-			} else if( parameterType.isAssignableFrom(argumentType) ) {
-				compatibleSetter = method; // potential match, but keep looking for exact match
-			}
-			
-		}
+                Class<?> actualType = clazz;
+                Method compatibleSetter = null;
+                while(actualType != null && compatibleSetter == null) {
+                        Method[] methods = actualType.getDeclaredMethods();
+                        for(Method method: methods) {
+                                if (!setterName.equalsIgnoreCase(method.getName()) || method.getParameterTypes().length != 1) {
+                                        continue; // setter must have correct name and only 1 parameter
+                                }
+
+                                final Class<?> parameterType = method.getParameterTypes()[0];
+                                if (parameterType.equals(argumentType)) {
+                                        return method;
+                                } else if (parameterType.isAssignableFrom(argumentType)) {
+                                        compatibleSetter = method; // potential match, but keep looking for exact match
+                                }
+                        }
+                        actualType = actualType.getSuperclass();
+                }
```

Comment:
I thought that we only want to set visible methods for current bean class. This change could have some side effects. @kbilsted, what do you think about this?